policy/release/cve.rego

```diff
@@ -181,12 +184,72 @@ deny contains result if {
 	result := lib.result_helper(rego.metadata.chain(), [error])
 }
 
+# extracts the clair report attached to the image
+_clair_report := report if {
+	clair_task := lib.task_in_pipelinerun(_cve_scan_task_name)
+
+	tekton.is_trusted_task(clair_task)
+	some task_result in lib.task_results(clair_task)
+	task_result.name == _reports_result_name
+	reports := lib.unmarshal(task_result.value)
+
+	input_image := image.parse(input.image.ref)
+
+	report_image := object.union(input_image, {"digest": reports[input_image.digest]})
+	report_manifest := ec.oci.image_manifest(image.str(report_image))
+
+	some layer in report_manifest.layers
+	layer.mediaType == _report_oci_mime_type
+	report_blob := object.union(input_image, {"digest": layer.digest})
+
+	report := json.unmarshal(ec.oci.blob(image.str(report_blob)))
+}
+
+# maps vulnerabilities and reports the counts by category (patched/unpatched)
+# and severity
+_clair_vulnerabilities[category] := vulns if {
+	reported_vulnerabilities := _clair_report.vulnerabilities
+
+	some category, vulnerabilities in {
+		"vulnerabilities": [v |
+			some v in reported_vulnerabilities
+			v.fixed_in_version != ""
+		],
+		"unpatched_vulnerabilities": [v |
+			some v in reported_vulnerabilities
+			v.fixed_in_version = ""
+		],
+	}
+
+	vulns := {
+		"critical": _count_by_severity_outside_leeway(vulnerabilities, "critical"),
+		"high": _count_by_severity_outside_leeway(vulnerabilities, "high"),
+		"medium": _count_by_severity_outside_leeway(vulnerabilities, "medium"),
+		"low": _count_by_severity_outside_leeway(vulnerabilities, "low"),
+		"unknown": _count_by_severity_outside_leeway(vulnerabilities, "unknown"),
+	}
+}
+
+# counts the vulnerabilities with the given severity excluding vulnerabilities
+# within the leeway period
+_count_by_severity_outside_leeway(vulnerabilities, severity) := count([v |
+	some v in vulnerabilities
+	lower(v.normalized_severity) == severity
+	leeway_days := lib.rule_data("cve_leeway")[severity]
+	time.add_date(time.parse_rfc3339_ns(v.issued), 0, 0, leeway_days) < lib_time.effective_current_time_ns
+])
+
```

Comment:
Probably as future work, but it would be great to create policy rules that specify the CVE as the term so specific CVEs could be ignored. That might require rethinking most of the policy rules on this package, and that the reports result is always available.